terra_notebook_utils/blobstore/copy_client.py

```diff
@@ -34,7 +35,8 @@ def _copy_intra_cloud(src_blob: AnyBlob, dst_blob: AnyBlob):
     # compute the destination S3Etag on the fly.
     logger.debug(f"Starting intra-cloud {src_blob.url} to {dst_blob.url}")
     assert isinstance(src_blob, type(dst_blob))
-    dst_blob.copy_from(src_blob)  # type: ignore
+    for part_size in dst_blob.copy_from(src_blob):  # type: ignore
```

Comment:
I liked the simplicity of just calling `download()` or 'copy_from()` as a one-liner earlier.  If there a way to shuttle the `for` loop into those functions and avoid it here?

Reply:
Perhaps the current methods should be renamed to `download_iter` and `copy_from_iter`, and convenience one-liners can be introduced as such:
```
def download(self, path):
    for part in self.download_iter(path):
         pass
```

I think I'll do this.

Reply:
btw, copy_client would need to use the `_iter` versions since it needs to report progress.